### PR TITLE
Update CODEWONERS file

### LINKED ISSUE #794: Update CODEOWNERS to more clearly identify scopes of concern for developers
## Description of issue / requirement to address

We need to streamline the review process and defining responsibility for sub-modules will help with that. Developers are struggling to maintain oversight of a wide scope.

## Proposed solution

Update the CODEOWNERS file to clearly identify a subset of necessary reviewers.

## Alternative solutions

Ad hoc approach isn't working so well!

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

closes #794 

## Description
Updates CODEOWNERS

## Interface Changes

N/A

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by je-cook — APPROVED** [review]
I'm happy but I will wait for others to approve before merging

File: CODEOWNERS
Comment (hsaunders1904) [inline CODEOWNERS:14]: We talked about me becoming more familiar with the equilibria module, should I be on this list?

**Review by hsaunders1904 — COMMENTED** [review]
Looks fine to me, only one question

File: CODEOWNERS
Comment (CoronelBuendia) [reply CODEOWNERS:14]: We did, and then @je-cook said that people were struggling with scope of concern! We can talk about this though, for sure.

Comment (CoronelBuendia) [reply CODEOWNERS:14]: Also should probably add @ivanmaione to equilibria now with the fixed boundary solver on the way

Comment (hsaunders1904) [reply CODEOWNERS:14]: > We did, and then @je-cook said that people were struggling with scope of concern! We can talk about this though, for sure.

Happy to leave it for now. We can always add me in if/when I start working in that area

Comment (CoronelBuendia) [reply CODEOWNERS:14]: Might actually be a better idea to get you on magnetostatics.. come to think of it. Spread the load a bit better

**Review by je-cook — APPROVED** [review]

**Pull request merged**
